FirebaseSwiftUI/FirebaseFacebookSwiftUI/Sources/Views/FacebookButtonView.swift

```diff
@@ -0,0 +1,108 @@
+import AppTrackingTransparency
+import FacebookCore
+import FacebookLogin
+import FirebaseAuth
+import FirebaseAuthSwiftUI
+import SwiftUI
+
+@MainActor
+public struct FacebookButtonView {
+  @Environment(AuthService.self) private var authService
+  @State private var errorMessage = ""
+  @State private var showCanceledAlert = false
+  @State private var showUserTrackingAlert = false
+  @State private var limitedLogin = true
+
+  public init() {}
+
+  private var limitedLoginBinding: Binding<Bool> {
```

Comment:
This computed property does a lot. It is trying to sync with ATTrackingManager.trackingAuthorizationStatus, and also drives showUserTrackingAlert and limitedLogin.

We should simplify this.

Reply:
Here is an alternative solution that feels cleaner:

```
  @State private var showUserTrackingAlert = false
  @State private var trackingAuthorizationStatus: ATTrackingManager.AuthorizationStatus = .notDetermined

  var isLimitedLogin: Bool {
    trackingAuthorizationStatus != .authorized
  }

  public init() {
    _trackingAuthorizationStatus = State(initialValue: ATTrackingManager.trackingAuthorizationStatus)
  }

  func requestTrackingPermission() {
    ATTrackingManager.requestTrackingAuthorization { status in
      Task { @MainActor in
        trackingAuthorizationStatus = status
        if status != .authorized {
          showUserTrackingAlert = true
        }
      }
    }
  }
```

Reply:
updated to reduce scope of binding: [`0d5d8ba` (#1237)](https://github.com/firebase/FirebaseUI-iOS/pull/1237/commits/0d5d8baa0dc001c5c72203e6c184d90e9400a215)

Not sure if you agree but a binding was still needed to stop toggle from switching if limited login wasn't possible.